modules/tinymce/src/themes/silver/main/ts/ui/toolbar/button/ToolbarButtons.ts

```diff
@@ -120,9 +156,11 @@ const renderCommonStructure = (
         receiver.map((r) => Reflecting.config({
           channel: r,
           initialData: { icon, text },
-          renderComponents: (data: ButtonState, _state) => componentRenderPipeline([
-            data.icon.map((iconName) => renderIconFromPack(iconName, providersBackstage.icons)),
-            data.text.map((text) => renderLabel(text, ToolbarButtonClasses.Button, providersBackstage))
+          renderComponents: (_data: ButtonState, _state) => componentRenderPipeline([
+            // data.icon.map((iconName) => renderIconFromPack(iconName, providersBackstage.icons)),
```

Comment:
Remove the old code here.

Reply:
I am waiting for @ephox-mogran to review this code and see if this approach is the right approach or not. That is why I am keeping these comments until I get a review from him.

Reply:
I think we can see two options here 

* if we still need to respond to the channel `r` broadcasts, then instead of being reflecting, this should be receiving, and it should respond to the message by emitting the custom events: `updateMenuText` and `updateMenuIcon` with the data passed as part of the message
* if we don't need to respond to the channel `r` broadcasts, then we should just remove this reflecting altogether.

Either way, what we have here isn't likely to be correct. If it receives a broadcast, it's going to reset the text and icon back to the original, and clobber any of the subsequent replacing calls, based on no information (it isn't using the information from the reflecting message)